docs/en/advanced/Scalability-Tuning.md

```diff
@@ -5,4 +5,84 @@ nickname: Scalability Tuning
 group: Advanced
 priority: 0
 ---
-TODO(adit): Add scalability related tuning tips
+
+## Introduction
+
+Alluxio is a scalable distributed file system designed to handle many workers within a single cluster.
+Several parameters can be tuned to prevent the Alluxio master from getting overloaded. This page
+aims to enumerate a list of such parameters which should be tuned for the scale of your cluster.
+
+### Alluxio Master Configuration
+
+#### Heap Size
+
+The Alluxio master heap size directly controls the total number of files that can fit into the
+master memory. Control the heap size by setting JVM options for the Alluxio master and secondary
+master processes. For example, to set the heap size to `256GB`, modify `alluxio-env.sh` as follows:
+```properties
+ALLUXIO_MASTER_JAVA_OPTS+=" -Xmx256g "
+ALLUXIO_SECONDARY_MASTER_JAVA_OPTS+=" -Xmx256g "
+```
+
+Note: Each thread spawned by the master JVM requires off heap space determined by the thread stack
+size. When setting the heap size, ensure that you have enough memory allocated for off heap storage.
+
+#### Thread Pool Size
+An executor pool is used on the master to handle concurrent client requests. If you expect a large
+number of concurrent clients communicating with the master, tune the thread pool size by modifying
+the following properties. The actual thread count depends on the maximum concurrency expected on the
+cluster.
```

Comment:
Add a calculation of how many clients/workers correspond to how many threads required?